### PR TITLE
READ_DATABASE / READ_TABLE and passing the ownerName in HivePrivilegeObject.

### PR DESCRIPTION
**Pull request by xBis7**

### CODE REVIEW

**Comment by xBis7** [comment]
@GeorgeJahad

File: standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java [hunk 1]
```diff
@@ -294,7 +294,7 @@ static void prewarm(RawStore rawStore) {
               AggrStats aggrStatsAllButDefaultPartition = null;
               if (table.isSetPartitionKeys()) {
                 Deadline.startTimer("getPartitions");
-                partitions = rawStore.getPartitions(catName, dbName, tblName, Integer.MAX_VALUE);
+                partitions = rawStore.getPartitions(catName, dbName, tblName, -1);
```
Comment (mladjan-gadzic) [inline standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java:297]: Why is this change needed? It looks like it doesn't have to do with anything we wanted to achieve in this PR. Same on 623 line below.

**Comment by mladjan-gadzic** [comment]
@xBis7 thanks for the work! 

I've left minor comment inline, but aside from that it looks good to me. Before approving, I'd like to know whether this was tested manually?

File: standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java (hunk 1 above)
Comment (xBis7) [reply standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java:297]: Most of these changes have been backported. I haven't added them manually. I've added them with `git cherry-pick`.

**Comment by xBis7** [comment]
> I'd like to know whether this was tested manually?

Yes, but that's a work in progress. I keep testing them and making changes to get all 3, Ranger, Hive and the scripts, working together.

**Review by mladjan-gadzic — APPROVED** [review]

**Comment by xBis7** [comment]
@mladjan-gadzic All tests are passing. I'm merging it, so that I can use the commit SHA.

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/ReadDatabaseEvent.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/*
```
Comment (GeorgeJahad) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/ReadDatabaseEvent.java:1]: @xBis7 Exactly which commit did you copy this class from?

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/ReadTableEvent.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/*
```
Comment (GeorgeJahad) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/ReadTableEvent.java:1]: same as above

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/ReadDatabaseEvent.java (hunk 2 above)
Comment (xBis7) [reply ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/ReadDatabaseEvent.java:1]: These two classes are from this commit 

https://github.com/apache/hive/commit/1a1fd15e3daa69e6dd5816ec877c3e8634cb7718

I couldn't backport the commit and I added them manually.

Comment (GeorgeJahad) [reply ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/ReadDatabaseEvent.java:1]: @xBis7 ok, thanks.  I would have preferred this not to have been a squash merge.  It is very nice to see that most of the code was cherry picked

Comment (xBis7) [reply ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/ReadDatabaseEvent.java:1]: I think I can undo it. Either locally and then pushing directly to the branch or with a new PR that deletes this commit and pushes 3-4 new ones.
